llvm/lib/IR/Verifier.cpp

```diff
@@ -2259,6 +2260,13 @@ void Verifier::verifyFunctionAttrs(FunctionType *FT, AttributeList Attrs,
       CheckFailed(
           "invalid value for 'branch-target-enforcement' attribute: " + S, V);
   }
+
+  if (auto A = Attrs.getFnAttr("vector-function-abi-variant"); A.isValid()) {
```

Comment:
I would have preferred this check be added in a separate patch, but I guess the change is small enough.

Improvements can be made in a later patch (e.g. explaining parameter type mismatches instead of just a 'invalid name' error).